pkg/driver/trace_driver.go

```diff
@@ -215,12 +217,23 @@ func (d *Driver) invokeFunction(metadata *InvocationMetadata) {
 
 		record.Phase = int(metadata.Phase)
 		record.InvocationID = composeInvocationID(d.Configuration.TraceGranularity, metadata.MinuteIndex, metadata.InvocationIndex)
-		metadata.RecordOutputChannel <- record
+
+		if !d.Configuration.LoaderConfiguration.AsyncMode || record.AsyncResponseID == "" {
+			metadata.RecordOutputChannel <- record
+		} else {
+			record.TimeToSubmitMs = record.ResponseTime
+			d.AsyncRecords.Enqueue(record)
+		}
 
 		if !success {
 			log.Debugf("Invocation failed at minute: %d for %s", metadata.MinuteIndex, function.Name)
+
+			atomic.AddInt64(metadata.FailedCount, 1)
+			atomic.AddInt64(&metadata.FailedCountByMinute[metadata.MinuteIndex], 1)
```

Comment:
We have updates to these counters later, close to exit from the function. It seems that it will be executed twice now. This might be the reason for the unit test failure.